hw/formal/formal_conn.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# Script to run connectivity test using Jasper Gold
+#
+# Usage: To run connectivity test on system level modules, type
+#   formal_conn -f top_earlgrey.csv
+#
+# More options:
+# -top: which top_level module to run
+# -p:  provide core file path
+# -gui: run with GUI
+# -cov: run coverage
+# -f: path to the csv file
+# -t: choose which formal tool to use, current only support jaspergold
+#
+# Example:
+#   formal_conn -gui -cov -f top_earlgrey.csv
+#
+
+export TOP="top_earlgrey"
+gui=0
+tool="jg"
+export COV=0
+export CSV_PATH
+
+while [ "$1" != "" ]; do
+  case "$1" in
+    "-top")
+      shift
+      TOP=$1
+      ;;
+    "-p")
+      shift
+      CORE_PATH=$1
+      ;;
+    "-gui")
+      gui=1
+      echo "using GUI mode"
+      ;;
+    "-cov")
+      COV=1
+      echo "enable coverage"
+      ;;
+    "-t")
+      shift
+      tool=$1
+      echo "running in tool: $tool"
+      ;;
+    "-f")
+      shift
+      CSV_PATH=../../../conn_csvs/$1
+      echo "csv path: $CSV_PATH"
+      ;;
+  esac
```

Comment:
Should there be a `*)` error case here, maybe? I think that at the moment the script silently discards arguments that it doesn't understand, which might be a tad confusing.

Reply:
Thank you Rupert! That is a great suggestion. Yeah having this wildcard to catch error would be very helpful for debug.